src/tools/cephfs/cephfs-shell

```diff
@@ -1040,25 +1040,54 @@ sub-directories, files')
         """
         self.poutput(os.getcwd())
 
+    def complete_df(self, text, line, begidx, endidx):
+        """
+        auto complete of file name.
+        """
+        return self.complete_filenames(text, line, begidx, endidx)
+
+    df_parser = argparse.ArgumentParser(
+        description='Show information about the file system on which each File resides,\
+            or all file systems by default.')
+    df_parser.add_argument('file', help='Name of the file', type=str,
+                           nargs='*', default='')
+
+    @with_argparser(df_parser)
     def do_df(self, arglist):
         """
         Display the amount of available disk space for file systems
         """
-        for index, i in enumerate(list_items(cephfs.getcwd())[2:]):
-            if index == 0:
-                self.poutput('{:25s}\t{:5s}\t{:15s}{:10s}{}'.format(
-                    "1K-blocks", "Used", "Available", "Use%", "Stored on"))
-            if not is_dir_exists(i.d_name):
-                statfs = cephfs.statfs(i.d_name)
-                stat = cephfs.stat(i.d_name)
+        flist = []  # List of valid existing directory or file names
+
+        for files in arglist.file:
+            if is_dir_exists(files) or is_file_exists(files):
+                flist.append(files)
+            else:
+                print("df: {}: No such file or directory".format(files))
+
+        if not arglist.file:
+            flist = list_items(cephfs.getcwd())[2:]
+
+        if flist:
+            self.poutput('{:25s}\t{:5s}\t{:15s}{:10s}{}'.format(
+                "1K-blocks", "Used", "Available", "Use%", "Stored on"))
+
+            for i in flist:
+                if not arglist.file:
+                    i = i.d_name
+                statfs = cephfs.statfs(i)
```

Comment:
The simpler thing to do here is to do:

````
try:
    stat = cephfs.statfs(file)
except cephfs.OSError as e:
    print("could not statfs {}: {}".format(file, e.strerror), file=sys.stderr)
````

Right now the rest of cephfs-shell is mixing error output with expected otuput to stdout. We should fix that (in a separate issue/PR).

Reply:
I will create a new PR for it.